scripts/pre_commit_linter.py

```diff
@@ -580,6 +580,94 @@ def _check_bad_pattern_in_file(filename, content, pattern):
     return False
 
 
+def _check_bad_pattern_in_comments_docstrings(filename, content):
+    """Detects whether comments and docstrings end with a period.
+
+    Args:
+        filename: str. Name of the file.
+        content: str. Contents of the file.
+
+    Returns:
+        bool. True if there is bad pattern else false.
+    """
+    regexp_comment = r"#[^\n]*[^.,\n]\n( |\t)*(#(( |\t)*[A-Z])|[^#\s])"
+    message_comment = "There should be a period after comment end."
+    message_docstring = "There should be a period after docstring end."
+    punctuate = ['.',',',';',':']
+    excluded_words = ['utf','pylint:','http://']
+    excluded_dirs = ()
+    excluded_files = ()
```

Comment:
What happens when a method/function/class without any docstrings is checked? Does the regex take care of this?

Reply:
Yes, it does!